### PR TITLE
Add PSRP connection type

### PR DESCRIPTION
**Pull request by arjunanan6**
This change intends on adding the PowerShell remoting protocol (PSRP) type connection along with a test function.

related: #19739

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — COMMENTED** [review]
Some tips for better contribution:
- Use a good/clear title for your PR
- Add a description for your change even if you add the related issue
- Keep your PR as draft while it is not ready

**Comment by Taragolis** [comment]
> related: https://github.com/apache/airflow/issues/19739

Are you sure that this correct issue? Issue related to [apache-airflow-providers-microsoft-winrm](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-winrm/stable/index.html) but changes made in [apache-airflow-providers-microsoft-psrp](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-psrp/stable/index.html)

**Comment by Taragolis** [comment]
If this changes are correct, than I guess it more related to
- https://github.com/apache/airflow/issues/28790

**Comment by arjunanan6** [comment]
@hussein-awala I didn't even know you could do a draft, so thanks for pointing that out. My final version will have a better title, details, etc. 

@Taragolis It is kind-of related, I guess it can take down two stones at a time.

**Comment by arjunanan6** [comment]
@hussein-awala or @Taragolis do you see what I am doing wrong here? I added the connection-type in provider.yaml which appears to be where the provider info and conns are looked up from for the UI, but it does not show up.

File: airflow/providers/microsoft/psrp/hooks/psrp.py [hunk 1]
```diff
@@ -75,6 +75,11 @@ class PsrpHook(BaseHook):
     or by setting this key as the extra fields of your connection.
     """
 
+    conn_name_attr = "psrp_conn_id"
+    default_conn_name = "psrp_default"
+    conn_type = "psrp"
+    hook_name = "PSRP"
```
Comment (Taragolis) [inline airflow/providers/microsoft/psrp/hooks/psrp.py]:
```suggestion
    hook_name = "PowerShell Remoting Protocol"
```

File: airflow/providers/microsoft/psrp/provider.yaml [hunk 2]
```diff
@@ -57,3 +57,7 @@ hooks:
   - integration-name: Windows PowerShell Remoting Protocol
     python-modules:
       - airflow.providers.microsoft.psrp.hooks.psrp
+
+connection-types:
+  - hook-class-name: airflow.providers.microsoft.psrp.hooks.PsrpHook
```
Comment (Taragolis) [inline airflow/providers/microsoft/psrp/provider.yaml]: Wrong path to hook

```suggestion
  - hook-class-name: airflow.providers.microsoft.psrp.hooks.psrp.PsrpHook
```

**Comment by Taragolis** [comment]
Problem with incorrect path to the PsrpHook, this could be found in case of running [Static Code Checks](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)

```console
Finished check: Found 0 errors in 1 checked items

Found 1 error in providers
Error: The `airflow.providers.microsoft.psrp.hooks.PsrpHook` object in connection-types list in airflow/providers/microsoft/psrp/provider.yaml does not exist or is not a class: module 'airflow.providers.microsoft.psrp.hooks' has no attribute 'PsrpHook'
```

Some other useful links for which might help you for contribution to Airflow
- [Contributor's Quick Start](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst#contributors-quick-start)
- [Airflow Breeze](https://github.com/apache/airflow/blob/main/BREEZE.rst)
- [CONTRIBUTING.rst](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst)

**Comment by arjunanan6** [comment]
@Taragolis Yikes, what a terrible oversight. I saw the static checks but did not check it properly enough. Thank you!

**Comment by arjunanan6** [comment]
@malthe Haven't had any time of late, but let me see if I can finish it up this week.

**Comment by arjunanan6** [comment]
@Taragolis @malthe  what do you think about the test I've just added?

**Comment by malthe** [comment]
@arjunanan6 you don't need to actually invoke any command; instead, enter into the hook context:
```python
with PsrpHook(psrp_conn_id=self.conn_id):
    pass
```
This will raise an exception if there is a connection error.

**Comment by arjunanan6** [comment]
> @arjunanan6 you don't need to actually invoke any command; instead, simply enter into the hook context:
> 
> ```python
> with PsrpHook(psrp_conn_id=self.conn_id):
>     pass
> ```
> 
> This will raise an exception if there is a connection error.
> 
> It works because entering the hook will in turn [enter into the runspace pool context](https://github.com/jborean93/pypsrp/blob/d2a3eca257230e40354d39ca008639992f8495a4/src/pypsrp/powershell.py#L257), which automatically connects it to the remote machine, establishing a connection.

Ah interesting, so I overdid it then. I'll make a change shortly and push it.

**Comment by malthe** [comment]
@arjunanan6 now, to test this, you would build on the existing suite:
```python
class TestPsrpHook:
    ....
    def test_get_conn(self, runspace_pool, *mocks):
        with PsrpHook(CONNECTION_ID):
            # Assert that the `runspace_pool` has been entered into, i.e. opened.
```
Something along the lines of
```python
assert runspace_pool.return_value.__enter__.mock_calls == [call()]
```

**Comment by arjunanan6** [comment]
@malthe Awesome, thanks for that! I'll get on it.

**Comment by arjunanan6** [comment]
I'm aware of the failing tests, I will look into it asap!

**Comment by arjunanan6** [comment]
@Taragolis Sorry about the extreme delay, I had some stuff going on. I think this is now ready for a review. Could you take a look when you have time? Thank you.

**Review by malthe — COMMENTED** [review]
Looks good to me

**Review by malthe — APPROVED** [review]

**Pull request merged**